nym-wallet/src/pages/bonding/node-settings/settings-pages/general-settings/index.tsx

```diff
@@ -1,41 +1,53 @@
 import React, { useState } from 'react';
 import { Box, Button, Divider, Grid } from '@mui/material';
 import { TBondedMixnode, TBondedGateway } from '../../../../../context/bonding';
-import { InfoSettings } from './InfoSettings';
+import { GeneralMixnodeSettings } from './GeneralMixnodeSettings';
 import { ParametersSettings } from './ParametersSettings';
-import { isMixnode } from 'src/types';
+import { isGateway, isMixnode } from 'src/types';
+import { GeneralGatewaySettings } from './GeneralGatewaySettings';
 
-const nodeGeneralNav = ['Info', 'Parameters'];
+const makeGeneralNav = (bondedNode: TBondedMixnode | TBondedGateway) => {
+  const navItems = ['Info'];
+  if (isMixnode(bondedNode)) {
+    navItems.push('Parameters');
+  }
+
+  return navItems;
+};
 
 export const NodeGeneralSettings = ({ bondedNode }: { bondedNode: TBondedMixnode | TBondedGateway }) => {
-  const [settingsCard, setSettingsCard] = useState<string>(nodeGeneralNav[0]);
+  const [navSelection, setNavSelection] = useState<number>(0);
   //TODO: Check what happens with a gateway
   return (
     <Box sx={{ pl: 3, pt: 3 }}>
       <Grid container direction="row" spacing={3}>
         <Grid item container direction="column" xs={3}>
-          {nodeGeneralNav.map((item) => (
+          {makeGeneralNav(bondedNode).map((item, index) => (
             <Button
               size="small"
               sx={{
                 fontSize: 14,
-                color: settingsCard === item ? 'primary.main' : 'inherit',
+                color: navSelection === index ? 'primary.main' : 'inherit',
                 justifyContent: 'start',
                 ':hover': {
                   bgcolor: 'transparent',
                   color: 'primary.main',
                 },
               }}
               key={item}
-              onClick={() => setSettingsCard(item)}
+              onClick={() => setNavSelection(index)}
             >
               {item}
             </Button>
           ))}
         </Grid>
         <Divider orientation="vertical" flexItem />
-        {settingsCard === nodeGeneralNav[0] && <InfoSettings bondedNode={bondedNode} />}
-        {settingsCard === nodeGeneralNav[1] && isMixnode(bondedNode) && <ParametersSettings bondedNode={bondedNode} />}
+        {navSelection === 0 && isMixnode(bondedNode) ? (
+          <GeneralMixnodeSettings bondedNode={bondedNode} />
+        ) : navSelection === 0 && isGateway(bondedNode) ? (
+          <GeneralGatewaySettings bondedNode={bondedNode} />
+        ) : null}
+        {navSelection === 1 && isMixnode(bondedNode) && <ParametersSettings bondedNode={bondedNode} />}
```

Comment:
@fmtabbara I think, I prefer the previous logic, even tho, the constants file seems more complex, here is easier to read I think. But both are right.

Reply:
Yes good point. I will maybe come back another time to make them the same format.